### ISSUE TITLE
Support use of django @override_settings to modify SPECTACULAR_SETTINGS in tests

### ISSUE THREAD

**Issue by AlexChalk** [tfranzel/drf-spectacular#719]
**Describe the bug**
Django gives us an `override_settings` decorator to test applications against different configurations: https://docs.djangoproject.com/en/4.0/topics/testing/tools/#django.test.override_settings.

I'm trying to use this to test the `spectacular` manage.py command against two different configurations, one for generating internal docs, and another for public_facing docs. 

Spectacular ignores the command and just uses my default (public-facing) configuration each time.

**To Reproduce**
Here are my example tests. Note that they are regression tests against pre-existing json files:
```
class DocsGenerationTest(TestCase):
    def test_public_facing_command_output_snapshot(self):
        out = StringIO()
        snapshot = Path("openapi_specs/public_facing.json").read_text()

        call_command(
            "spectacular",
            format="openapi-json",
            validate=True,
            fail_on_warn=True,
            stdout=out,
        )

        assert json.loads(snapshot) == json.loads(
            out.getvalue()
        ), SNAPSHOT_CHANGE_MESSAGE

    @override_settings(SPECTACULAR_SETTINGS=INTERNAL_SPECTACULAR_SETTINGS)
    def test_internal_command_output_snapshot(self):
        out = StringIO()
        snapshot = Path("openapi_specs/internal.json").read_text()

        call_command(
            "spectacular",
            format="openapi-json",
            validate=True,
            fail_on_warn=True,
            stdout=out,
        )

        assert json.loads(snapshot) == json.loads(
            out.getvalue()
        ), SNAPSHOT_CHANGE_MESSAGE
```

To reproduce, write a second test like this pointing to a `SPECTACULAR_SETTINGS` configuration different from your default; the default configuration is used anyway.

**Expected behavior**
I'd like `@override_settings` to successfully override my `SPECTACULAR_SETTINGS` and let me test the generated output for different configurations. 

Let me know if you need further info/if I can help. Thanks!

**Comment by tfranzel** [comment]
Hi @AlexChalk,

I see your point but I am unsure whether this is actually supposed to apply to 3rd party app settings too. I think not. Settings are loaded from the settings file into an `APISettings` object. We had to slighly adapt that class.  I believe that decorator explicitly targets that DRF object and nothing else. Their doc is also very explicit about that:

```
    This is an internal class that is only compatible with settings namespaced
    under the REST_FRAMEWORK name. It is not intended to be used by 3rd-party
    apps, and test helpers like `override_settings` may not work as expected.
```
https://github.com/encode/django-rest-framework/blob/b1004a47334a0dd1929e6d50b8f7ff6badc959f4/rest_framework/settings.py#L194

On our side, we have made heavy use of `patch` with good results in our tests. This should reliably work for you too 

```python
@mock.patch(
    'drf_spectacular.settings.spectacular_settings.POSTPROCESSING_HOOKS',
    [camelize_serializer_fields]
)
```

**Comment by AlexChalk** [comment]
Thanks @tfranzel, this is a fine solution if we can get it to work! Right now I can't though:

```
INTERNAL_SPECTACULAR_API_SETTINGS = APISettings(
    user_settings=INTERNAL_SPECTACULAR_SETTINGS,
    defaults=settings.SPECTACULAR_DEFAULTS,
    import_strings=settings.IMPORT_STRINGS,
)

...
    @mock.patch(
        "drf_spectacular.settings.spectacular_settings", INTERNAL_SPECTACULAR_API_SETTINGS
    )
```

This still doesn't override the settings correctly for me?

**Comment by AlexChalk** [comment]
(Also I'm tying to your implementation in the tests more than I'd like here.)

**Comment by tfranzel** [comment]
> INTERNAL_SPECTACULAR_API_SETTINGS = APISettings(

this is not what I suggested there. In fact your are using DRF's version instead of ours. Apart from that I do not know if that actually works. We only do it per variable in our tests. your mileage may vary.

Loading settings and overriding stuff is a tricky business, the order in which things are imported and used do matter. 

> (Also I'm tying to your implementation in the tests more than I'd like here.)

the fact that `@override_settings` explicitly only works for DRF was not my design choice. The whole settings thing in Django is a bit messy imho.

**Comment by AlexChalk** [comment]
@tfranzel Yep I'm trying to set them all at once, if that's not something you've done in tests I guess I'll experiment some more.

> In fact your are using DRF's version instead of ours. 

We're currently on 0.15.1, maybe this has changed since, but in `drf_spectacular/settings.py` `spectacular_settings` is exported as an instance of the DRF class.

**Comment by tfranzel** [comment]
I would strongly recommend updating to the most recent version. there have been tons of improvement all over the place since then. 

as a matter of fact we added a context manager `patched_settings` to be able to patch the settings on a view level. You may find this useful for your tests, iff you do not intend to use that feature normally.

https://github.com/tfranzel/drf-spectacular/blob/b3e00be95eb9087af4bd2189ca37d87e8b6c9f6a/drf_spectacular/settings.py#L244

**Comment by AlexChalk** [comment]
Thanks @tfranzel, `patched_settings` actually looks great for our use case!
